community/kernel/src/main/java/org/neo4j/kernel/impl/coreapi/PropertyContainerLocker.java

```diff
@@ -36,71 +36,70 @@
 {
     public Lock exclusiveLock( Supplier<Statement> stmtSupplier, PropertyContainer container )
     {
-        try(Statement statement = stmtSupplier.get())
+        Statement statement = stmtSupplier.get();
+
+        if ( container instanceof Node )
         {
-            if(container instanceof Node )
+            statement.readOperations().acquireExclusive( ResourceTypes.NODE, ((Node) container).getId() );
+            return new CoreAPILock( stmtSupplier, ResourceTypes.NODE, ((Node) container).getId() )
             {
-                statement.readOperations().acquireExclusive( ResourceTypes.NODE, ((Node) container).getId() );
-                return new CoreAPILock(stmtSupplier, ResourceTypes.NODE, ((Node) container).getId())
+                @Override
+                void release( Statement statement, ResourceType type, long resourceId )
                 {
-                    @Override
-                    void release( Statement statement, ResourceType type, long resourceId )
-                    {
-                        statement.readOperations().releaseExclusive( type, resourceId );
-                    }
-                };
-            }
-            else if(container instanceof Relationship )
+                    statement.readOperations().releaseExclusive( type, resourceId );
+                }
+            };
+        }
+        else if ( container instanceof Relationship )
+        {
+            statement.readOperations()
+                    .acquireExclusive( ResourceTypes.RELATIONSHIP, ((Relationship) container).getId() );
+            return new CoreAPILock( stmtSupplier, ResourceTypes.RELATIONSHIP, ((Relationship) container).getId() )
             {
-                statement.readOperations().acquireExclusive( ResourceTypes.RELATIONSHIP, ((Relationship) container).getId() );
-                return new CoreAPILock(stmtSupplier, ResourceTypes.RELATIONSHIP, ((Relationship) container).getId())
+                @Override
+                void release( Statement statement, ResourceType type, long resourceId )
                 {
-                    @Override
-                    void release( Statement statement, ResourceType type, long resourceId )
-                    {
-                        statement.readOperations().releaseExclusive( type, resourceId );
-                    }
-                };
-            }
-            else
-            {
-                throw new UnsupportedOperationException( "Only relationships and nodes can be locked." );
-            }
+                    statement.readOperations().releaseExclusive( type, resourceId );
+                }
+            };
+        }
+        else
+        {
+            throw new UnsupportedOperationException( "Only relationships and nodes can be locked." );
         }
+
     }
 
     public Lock sharedLock( Supplier<Statement> stmtProvider, PropertyContainer container )
```

Comment:
This method is also used by the `TopLevelTransaction` or `PlaceboTransaction` in order to implement [`Transaction.acquireReadLock(PropertyContainer)`](https://github.com/neo4j/neo4j/blob/3.0/community/graphdb-api/src/main/java/org/neo4j/graphdb/Transaction.java#L171).

The same reasoning applies here as for `exclusiveLock(...)`.